FreeRTOS_Stream_Buffer.c

```diff
@@ -209,31 +209,13 @@ BaseType_t xStreamBufferLessThenEqual( const StreamBuffer_t * pxBuffer,
                                        const size_t uxLeft,
                                        const size_t uxRight )
 {
-    BaseType_t xReturn;
+    BaseType_t xReturn = pdFALSE;
     size_t uxTail = pxBuffer->uxTail;
 
-    /* Returns true if ( uxLeft < uxRight ) */
-    if( ( ( ( uxLeft < uxTail ) ? 1U : 0U ) ^ ( ( uxRight < uxTail ) ? 1U : 0U ) ) != 0U )
+    /* Returns true if ( uxLeft <= uxRight ) */
+    if( ( uxLeft - uxTail ) <= ( uxRight - uxTail ) )
```

Comment:
We need to update our unit-tests so that we check that this change doesn't break anything.